compat/src/render.js

```diff
@@ -101,113 +101,125 @@ function isDefaultPrevented() {
-		if (
-			type == 'select' &&
-			normalizedProps.multiple &&
-			Array.isArray(normalizedProps.value)
+		let lowerCased = i.toLowerCase();
+		if (i === 'defaultValue' && 'value' in props && props.value == null) {
+			// `defaultValue` is treated as a fallback `value` when a value prop is present but null/undefined.
+			// `defaultValue` for Elements with no value prop is the same as the DOM defaultValue property.
+			i = 'value';
+		} else if (i === 'download' && value === true) {
+			// Calling `setAttribute` with a truthy value will lead to it being
+			// passed as a stringified value, e.g. `download="true"`. React
+			// converts it to an empty string instead, otherwise the attribute
+			// value will be used as the file name and the file will be called
+			// "true" upon downloading it.
+			value = '';
+		} else if (lowerCased === 'ondoubleclick') {
+			i = 'ondblclick';
+		} else if (
+			lowerCased === 'onchange' &&
+			(type === 'input' || type === 'textarea') &&
+			!onChangeInputType(props.type)
 		) {
-			// forEach() always returns undefined, which we abuse here to unset the value prop.
-			normalizedProps.value = toChildArray(props.children).forEach(child => {
-				child.props.selected =
-					normalizedProps.value.indexOf(child.props.value) != -1;
-			});
+			lowerCased = i = 'oninput';
+		} else if (lowerCased === 'onfocus') {
+			i = 'onfocusin';
+		} else if (lowerCased === 'onblur') {
+			i = 'onfocusout';
+		} else if (ON_ANI.test(i)) {
+			i = lowerCased;
+		} else if (type.indexOf('-') === -1 && CAMEL_PROPS.test(i)) {
+			i = i.replace(CAMEL_REPLACE, '-$&').toLowerCase();
+		} else if (value === null) {
+			value = undefined;
 		}
 
-		// Adding support for defaultValue in select tag
-		if (type == 'select' && normalizedProps.defaultValue != null) {
-			normalizedProps.value = toChildArray(props.children).forEach(child => {
-				if (normalizedProps.multiple) {
-					child.props.selected =
-						normalizedProps.defaultValue.indexOf(child.props.value) != -1;
-				} else {
-					child.props.selected =
-						normalizedProps.defaultValue == child.props.value;
-				}
-			});
+		// Add support for onInput and onChange, see #3561
+		// if we have an oninput prop already change it to oninputCapture
+		if (lowerCased === 'oninput') {
+			i = lowerCased;
+			if (normalizedProps[i]) {
+				i = 'oninputCapture';
+			}
 		}
 
-		vnode.props = normalizedProps;
+		normalizedProps[i] = value;
+	}
+
+	// Add support for array select values: <select multiple value={[]} />
+	if (
+		type == 'select' &&
+		normalizedProps.multiple &&
+		Array.isArray(normalizedProps.value)
+	) {
+		// forEach() always returns undefined, which we abuse here to unset the value prop.
+		normalizedProps.value = toChildArray(props.children).forEach(child => {
+			child.props.selected =
+				normalizedProps.value.indexOf(child.props.value) != -1;
+		});
+	}
 
-		if (props.class != props.className) {
```

Comment:
What happened before:

- props.class === undefined && props.className !== undefined --> we would make className enumerable and make class === className
- props.class !== undefined && props.className === undefined --> we would make className not enumerable and make className return props.class
- props.class !== undefined && props.className !== undefined --> we would make className enumerable and make class === className

Position 1 and 3 are different because we are just returning the value for `className` rather than the one for class, I don't think this matters.
Position 2 is different because we aren't introducing a getter for `className` to return `class`, I don't think this matters

CC @marvinhagemeister

Reply:
Thanks for checking. That sounds reasonable. I'm good with merging the PR 👍

Reply:
Yep that seems reasonable to me. I think the dynamic assignment of `enumerable` here was from when we used to make `class` a descriptor too.